### PR TITLE
Add color icons

### PR DESCRIPTION
**Pull request by tcwatson**
Adds a setting that allows the user to choose between beige and color icons. Stores all icons locally and remove dependencies on the Guild Wars 2 API.

### CODE REVIEW

File: Module.cs [hunk 1]
```diff
@@ -86,30 +99,22 @@ protected override void Initialize()
...
             );
 
-            var specializationClient = webApiClient.Specializations;
-            var eliteSpecs = await specializationClient.ManyAsync(Specialization.EliteCodes);
-
-            _specializationIcons = eliteSpecs.ToDictionary(
-                keySelector: (spec) => (uint)spec.Id,
-                (spec) => GameService.Content.GetRenderServiceTexture(spec.ProfessionIconBig)
+            _specializationIcons = Specialization.EliteCodes.ToDictionary(
+                keySelector: (spec) => (uint)spec,
```
Comment (sonicolasj) [inline Module.cs]: I think the cast can be removed, as spec is already `uint`

File: PlayerCollection.cs [hunk 2]
```diff
@@ -101,6 +101,14 @@ public void ClearFormerPlayers()
             }
         }
 
+        public void SwapIcons()
```
Comment (sonicolasj) [inline PlayerCollection.cs]: No need to have multiple names for the same action!

```suggestion
        public void RefreshIcons()
```

File: Specialization.cs [hunk 3]
```diff
@@ -57,6 +57,19 @@ public static string GetCoreName(uint core)
             };
         }
 
+        public static readonly IReadOnlyCollection<uint> ProfessionCodes = new uint[]
```
Comment (sonicolasj) [inline Specialization.cs:60]: Nitpick: I think it should be similar to the initialization below (`public static readonly IReadOnlyCollection<int> EliteCodes = new[]`), so either use `new[]` here, or expand to `new uint[]` below!

File: Specialization.cs [hunk 4]
```diff
@@ -95,5 +108,22 @@ public static string GetCoreName(uint core)
...
+
+        public static string GetCoreIconPath(uint core, bool useTangoIcons)
+        {
+            var folder = useTangoIcons ? "professions_tango" : "professions_beige";
+            return @$"{folder}\{GetCoreName(core)}.png";
+        }
+
+        private static string GetEliteName(uint elite)
+        {
+            return GetEliteName(elite, uint.MaxValue); //core uint not needed for elites
```
Comment (sonicolasj) [inline Specialization.cs]:
```suggestion
            return GetEliteName(elite, default); //core uint not needed for elites
```

File: Specialization.cs [hunk 5]
```diff
@@ -95,5 +108,22 @@ public static string GetCoreName(uint core)
...
+        {
+            var folder = useTangoIcons ? "professions_tango" : "professions_beige";
+            return @$"{folder}\{GetEliteName(elite)}.png";
+        }
+
+        public static string GetCoreIconPath(uint core, bool useTangoIcons)
+        {
+            var folder = useTangoIcons ? "professions_tango" : "professions_beige";
+            return @$"{folder}\{GetCoreName(core)}.png";
+        }
```
Comment (sonicolasj) [inline Specialization.cs:123]: I think these two functions would be better extracted elsewhere, as `GetIconPath(string iconName, bool useTangoIcons)`

**Review by sonicolasj — APPROVED** [review]
Really solid PR!

Some nitpicks here and there (tbh they can safely be ignored)

Just a passing thought: I'm not sure `tango`/`beige` is a great naming for the icons, instead of `colored`/`monochrome`?

File: Module.cs (hunk 1 above)
Comment (tcwatson) [reply Module.cs]: Had to change the type of `IReadOnlyCollection<int> EliteCodes` to `IReadOnlyCollection<uint> EliteCodes` to get rid of the cast

File: Specialization.cs (hunk 5 above)
Comment (tcwatson) [reply Specialization.cs:123]: Agreed, but not sure where to extract them to. Maybe better after the MVP refactor?

**Pull request merged**